apps/script/HoC2023AiGenerateWeights.py

```diff
@@ -2,157 +2,149 @@
 import spacy
 import json
 
-# Load the spaCy model
-nlp = spacy.load("en_core_web_md")
+# Load the spaCy natural language processing model - English tokenizer, tagger, parser and NER (named entity recognition)
+nlp = spacy.load("en_core_web_lg")
 
-# Define your "id" (emojis) and foreground/background/palette lists
-# For items like backgrounds/foregrounds, certain values were adjusted to reflect the most context-rich but open-ended word within the phrase
-# e.g. "ripples_random" is adjusted to "ripples", "smiling_poop" -> "poop", etc.
-# This was done due to spacy interpreting underscored phrases as a unique word and phrases such as "smiling poop" not correlating highly with most anything within a internet scraped corpus
-id = ["poopy", "romantic", "party", "silly", "sparkle", "happy", "magic", "spooky", "cute", "funky", "wavy", "lights", "rainbow", "robot", "chaotic", "disco", "zen", "fast", "evil", "cold", "cosmic", "sad", "black-and-white", "warm", "cool"]
-palettes = ["cool", "electronic", "ice cream", "neon", 'rave', "tropical","vintage",'warm', 'greyscale', 'sky', 'ocean', 'sunrise', 'sunset', 'spring', 'summer', 'autumn', 'winter', 'twinkling', 'rainbow', 'roses']
-backgrounds = ['circles',
-    'color',
-    'diamonds',
-    'disco',
-    'fireworks',
-    'swirl',
-    'kaleidoscope',
-    'lasers',
-    'splatter',
-    'rainbow',
-    'snowflakes',
-    'text',
-    'galaxy',
-    'sparkles',
-    'spiral',
-    'disco',
-    'stars',
-    'music',
-    'ripples',
-    'random',
-    'quads',
-    'flowers',
-    'squiggles',
-    'stars',
-    'petals',
-    'clouds',
-    'grid',
-    'starburst',
-  ]
+emoji_ids = ["poopy", "romantic", "party", "silly", "sparkle", "happy", "magic", "spooky", "cute", "funky", "wavy", "lights", "rainbow", "robot", "chaotic", "disco", "zen", "fast", "evil", "cold", "cosmic", "sad", "black-and-white", "warm", "cool"]
 
-foregrounds = [
-    'bubbles',
-    'confetti',
-    'hearts',
-    'music',
-    'pineapples',
-    'pizzas',
-    'poop',
-    'rain',
-    'rainbows',
-    'smile',
-    'spotlight',
-    'lights',
-    'tacos',
-    'emojis',
-    'hearts',
-    'stars',
-    'paint',
-  ]
+# We rename foreground/background effects and palettes to better reflect the
+# actual output of the effect or color. Below, we store them in a map (python_name: blockly_id).
 
-palettedict = {}
-bgdict = {}
-fgdict = {}
+# color_palettes dictionary contains python name as key and blockly id as value.
```

Comment:
Can we define "python name" somewhere?

Reply:
Now more explicitly defined in comment above.